go/bss-core/drivers/clear_pending_tx.go

```diff
@@ -159,7 +160,20 @@ func SignClearingTx(
 	}
 
 	gasFeeCap := txmgr.CalcGasFeeCap(head.BaseFee, gasTipCap)
-	tx := CraftClearingTx(walletAddr, nonce, gasFeeCap, gasTipCap)
+
+	gasLimit, err := l1Client.EstimateGas(ctx, ethereum.CallMsg{
+		From:      walletAddr,
+		To:        &walletAddr,
+		GasFeeCap: gasFeeCap,
+		GasTipCap: gasTipCap,
+		Value:     nil,
+		Data:      nil,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	tx := CraftClearingTx(walletAddr, nonce, gasFeeCap, gasTipCap, gasLimit)
 
```

Comment:
There could be a trace log statement that has all of the fields of the tx in case there are issues in the future